src/js/utils/spec-helpers/ajax-helpers.js

```diff
@@ -17,204 +28,264 @@ define(['sinon', 'underscore', 'URI'], function(sinon, _, URI) {
+    expectPostRequest = function(requests, url, body) {
+        var request = currentRequest(requests);
+        expect(request.readyState).toEqual(XML_HTTP_READY_STATES.OPENED);
         expect(request.url).toEqual(url);
         expect(request.method).toEqual('POST');
         expect(_.difference(request.requestBody.split('&'), body.split('&'))).toEqual([]);
     };
 
     /**
+     * Verify that the HTTP request was marked as reset, and then skip it.
+     *
+     * Note: this is typically used when code has explicitly canceled a request
+     * after it has been sent. A good example is when a user chooses to cancel
+     * a slow running search.
+
+     * @param {object} requests an array of fired sinon requests
+     */
+    skipResetRequest = function(requests) {
+        var request = currentRequest(requests);
+        expect(request.readyState).toEqual(XML_HTTP_READY_STATES.UNSENT);
+        requests.currentIndex++;
+    };
+
+    /**
      * Respond to a request with JSON.
      *
-     * @param requests an array of fired sinon requests
-     * @param jsonResponse an object to be serialized to the response
-     * @param requestIndex the expected index of the request in the
-     *     array (higher indices occur later).  Defaults to
-     *     requests.length - 1.
+     * @param {object} requests an array of fired sinon requests
+     * @param {object} jsonResponse an object to be serialized to the response
      */
-    respondWithJson = function(requests, jsonResponse, requestIndex) {
-        if (_.isUndefined(requestIndex)) {
-            requestIndex = requests.length - 1;
-        }
-        requests[requestIndex].respond(200,
-            { 'Content-Type': 'application/json' },
+    respondWithJson = function(requests, jsonResponse) {
+        var request = currentRequest(requests);
+        request.respond(200,
+            {'Content-Type': 'application/json'},
             JSON.stringify(jsonResponse));
+        requests.currentIndex++;
     };
 
     /**
      * Respond to a request with an error status code and a JSON
      * payload.
      *
-     * @param requests an array of fired sinon requests
-     * @param statusCode the HTTP status code of the response
+     * @param {object} requests an array of fired sinon requests
+     * @param {int} statusCode the HTTP status code of the response
      *     (defaults to 500)
-     * @param jsonResponse an object to be serialized to the response
-     * @param requestIndex the expected index of the request in the
-     *     array (higher indices occur later).  Defaults to
-     *     requests.length - 1.
-     */
-    respondWithError = function(requests, statusCode, jsonResponse, requestIndex) {
-        if (_.isUndefined(requestIndex)) {
-            requestIndex = requests.length - 1;
-        }
+     * @param {object} jsonResponse an object to be serialized to the response
+     */
+    respondWithError = function(requests, statusCode, jsonResponse) {
```

Comment:
I like that `statusCode` and `jsonResponse` are optional, but this construct enforces that the consumer must supply `statusCode` if they're going to supply `jsonResponse`. How about moving both into an `options` hash?